gammapy/spectrum/flux_point.py

```diff
@@ -843,20 +845,30 @@ def __init__(
         self.sigma_ul = sigma_ul
         self.reoptimize = reoptimize
         self.source = source
+        self.fit = Fit(self.datasets)
 
     def _freeze_parameters(self):
         # freeze other parameters
         for par in self.datasets.parameters:
             if par is not self.model.norm:
                 par.frozen = True
 
+    def _freeze_empty_background(self):
+        from ..cube import MapDataset
+
+        counts_all = self.estimate_counts()["counts"]
+
+        for counts, dataset in zip(counts_all, self.datasets.datasets):
+            if isinstance(dataset, MapDataset) and counts == 0:
```

Comment:
Will a `MapDataset` always have a `background_model`? One could also check for `hasattr(dataset, "background_model")`.

Reply:
Done